lib/storage/azure.js

```diff
@@ -170,34 +171,75 @@ module.exports = function() {
           if (err) {
             return cb(err);
           }
+
           blobSvcs.push({
             svc: svc,
             container: cluster.container || options.container
           });
+
           return cb();
         });
       }, callback);
     },
 
-    copyIn: function(localPath, path, options, callback) {
-      // TODO be more selective about when we do this
-      gzipme(localPath, false, 'best', function() {
-        path = path[0] === '/' ? path.slice(1) : path;
-        async.each(blobSvcs, function(blobSvc, cb) {
-          createContainerBlob(blobSvc, path, localPath, function(err) {
-            if (err) {
-              __log('! Cannot create blob in container : ' + err);
+    cleanupStreams: function (inputStream, outputStream, tempPath, err) {
+      async.parallel({
+        unlink: cb => {
+          removeLocalBlob(tempPath, cb);
+        },
+
+        closeReadStream: cb => {
+          inputStream.end(cb);
+        },
+
+        closeWriteStream: cb => {
+          outputStream.end(cb);
+        }
+      }, cleanupError => {
+        console.log("Error cleaning up from error", err, cleanupError);
```

Comment:
You never invoke the callback...? The callback should be passed to cleanupStreams and invoked at the end.